packages/typespec-azure-resource-manager/lib/Legacy/operations.tsp

```diff
@@ -0,0 +1,290 @@
+  >(
+    ...ParentParameters,
+    ...ResourceTypeParameter,
+    ...Parameters,
+    ...BodyParam,
+  ): Response | ErrorType;
+
+  /**
+   * Delete a resource asynchronously
+   * @template Resource The resource being deleted
+   * @template LroHeaders The lro headers for the operation
+   * @template Parameters Optional. Additional parameters after the path parameters
+   * @template Response Optional. The success response(s) for the delete operation
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-response-body" "Valid"
+  @autoRoute
+  @doc("Delete a {name}", Resource)
+  @delete
+  @deletesResource(Resource)
+  @armResourceDelete(Resource)
+  DeleteAsync<
+    Resource extends Azure.ResourceManager.CommonTypes.Resource,
+    LroHeaders extends TypeSpec.Reflection.Model = ArmLroLocationHeader &
+      Azure.Core.Foundations.RetryAfterHeader,
+    Parameters extends {} = {},
+    Response extends {} = ArmDeleteAcceptedLroResponse<LroHeaders> | ArmDeletedNoContentResponse
+  >(...ParentParameters, ...ResourceTypeParameter, ...Parameters): Response | ErrorType;
+
+  /**
+   * Delete a resource synchronously
+   * @template Resource The resource being deleted
+   * @template Parameters Optional. Additional parameters after the path parameters
+   * @template Response Optional. The success response(s) for the delete operation
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-response-body" "Valid"
+  @autoRoute
+  @doc("Delete a {name}", Resource)
+  @delete
+  @deletesResource(Resource)
+  @armResourceDelete(Resource)
+  DeleteSync<
+    Resource extends Azure.ResourceManager.CommonTypes.Resource,
+    Parameters extends {} = {},
+    Response extends {} = ArmDeletedResponse | ArmDeletedNoContentResponse
+  >(...ParentParameters, ...ResourceTypeParameter, ...Parameters): Response | ErrorType;
+
+  /**
+   * @dev Get a resource
+   * @template Resource The resource being read
+   * @template Parameters Optional. Additional parameters after the path parameters
+   * @template Response Optional. The success response for a get operation.
+   */
+  @autoRoute
+  @doc("Get a {name}", Resource)
+  @get
+  @readsResource(Resource)
+  @armResourceRead(Resource)
+  Read<
+    Resource extends Azure.ResourceManager.CommonTypes.Resource,
+    Parameters extends {} = {},
+    Response extends {} = ArmResponse<Resource>
+  >(...ParentParameters, ...ResourceTypeParameter, ...Parameters): Response | ErrorType;
+
+  /**
+   * List a resource
+   * @template Resource The resource being listed
+   * @template Parameters Optional. Additional parameters after the path parameters
+   * @template Response Optional. The response returned by the list
+   */
+  @autoRoute
+  @doc("List a {name}", Resource)
+  @get
+  @listsResource(Resource)
+  @segmentOf(Resource)
+  @armResourceList(Resource)
+  List<
+    Resource extends Azure.ResourceManager.CommonTypes.Resource,
```

Comment:
question, why are we using Azure.ResourceManager.CommonTypes.Resource in here instead of Foundations.Resource? for what I saw it is the same model, but I want to know why whe are using CommonTypes.Resource in here as it doesn't look consistent with the rest of the templates

Reply:
This is essentially because Foundations.Resource is an alias.  When the original templates were written, there was no common-types namespace, and we added the alias and moved the type into this namespace.  I am all for consistency, happy to use the alias (it shouldn't make any difference to the functionality).